net/dns/ini.go

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2021 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package dns
+
+import (
+	"regexp"
+	"strings"
+)
+
+// parseIni parses a basic .ini file, used for wsl.conf.
+func parseIni(data string) map[string]map[string]string {
+	sectionRE := regexp.MustCompile(`^\[([^]]+)\]\s*`)
+	kvRE := regexp.MustCompile(`^\s*(\w+)\s*=\s*([^#]*)\s*`)
```

Comment:
`;` can also be used for comments.

This actually looks pretty good: https://github.com/zieckey/goini

Reply:
Hmm. Two problems, goini expects: 1. comments start at the beginning of the line, 2. wsl.conf doesn't seem to be entirely standard?

This successfully turns off WSL resolve.conf generation here:

```
[network]
generateResolvConf = false # a comment
```

(but would be parsed incorrectly by goini I believe)

This does *not* turn off WSL resolv.conf generation on my machine:

```
[network]
generateResolvConf = false ; a comment
```

I would be happy to replace this code, but I don't think goini will quite work.

Reply:
I think ini files aren't completely standard. Alas. This domain specific parser is fine, I suppose, since all we really need is to detect what a particular boolean value in a particular section is set to.

I didn't catch that goini requires comments be at the start of the line. That's annoying.